File: src/main/java/nz/co/searchwellington/signin/SigninHandler.java
package nz.co.searchwellington.signin;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import nz.co.searchwellington.model.User;

import org.springframework.web.servlet.ModelAndView;

public interface SigninHandler {

	public ModelAndView getLoginView(HttpServletRequest request, HttpServletResponse response) throws Exception;
	
	public Object getExternalUserIdentifierFromCallbackRequest(HttpServletRequest request);
	public User getUserByExternalIdentifier(Object externalIdentifier);
	public void decorateUserWithExternalSigninIdentifier(User user, Object externalIdentifier);
	
}

File: src/main/scala/nz/co/searchwellington/signin/SigninHandler.scala
package nz.co.searchwellington.signin

import jakarta.servlet.http.{HttpServletRequest, HttpServletResponse}
import nz.co.searchwellington.model.User
import org.springframework.web.servlet.ModelAndView

trait SigninHandler {

  def getLoginView(request: HttpServletRequest, response: HttpServletResponse): ModelAndView
  def getExternalUserIdentifierFromCallbackRequest(request: HttpServletRequest): Option[Any]
  def getUserByExternalIdentifier(externalIdentifier: Any): Option[User]
  def decorateUserWithExternalSigninIdentifier(user: User, externalIdentifier: Any): User
}